src/main/java/bio/terra/cli/businessobject/resource/GcpDataprocCluster.java

```diff
@@ -0,0 +1,192 @@
+            .projectId(wsmObject.getResourceAttributes().getGcpDataprocCluster().getProjectId())
+            .region(wsmObject.getResourceAttributes().getGcpDataprocCluster().getRegion())
+            .name(wsmObject.getResourceAttributes().getGcpDataprocCluster().getClusterId())
+            .build();
+  }
+
+  /** Deserialize a cluster of the WSM client library create object to the internal object. */
+  public GcpDataprocCluster(GcpDataprocClusterResource wsmObject) {
+    super(wsmObject.getMetadata());
+    this.resourceType = Type.DATAPROC_CLUSTER;
+    this.clusterName =
+        ClusterName.builder()
+            .projectId(wsmObject.getAttributes().getProjectId())
+            .region(wsmObject.getAttributes().getRegion())
+            .name(wsmObject.getAttributes().getClusterId())
+            .build();
+  }
+
+  /**
+   * Add a GCP dataproc cluster as a referenced resource in the workspace. Currently unsupported.
+   */
+  public static GcpDataprocCluster addReferenced(CreateGcpDataprocClusterParams createParams) {
+    throw new UserActionableException(
+        "Referenced resources not supported for GCP dataproc clusters.");
+  }
+
+  /**
+   * Create a GCP dataproc cluster as a controlled resource in the workspace.
+   *
+   * @return the resource that was created
+   */
+  public static GcpDataprocCluster createControlled(CreateGcpDataprocClusterParams createParams) {
+    validateResourceName(createParams.resourceFields.name);
+
+    // call WSM to create the resource
+    GcpDataprocClusterResource createdResource =
+        WorkspaceManagerServiceGcp.fromContext()
+            .createControlledGcpDataprocCluster(Context.requireWorkspace().getUuid(), createParams);
+    logger.info("Created GCP dataproc cluster: {}", createdResource);
+
+    return new GcpDataprocCluster(createdResource);
+  }
+
+  /**
+   * Serialize the internal representation of the resource to the format for command input/output.
+   */
+  public UFGcpDataprocCluster serializeToCommand() {
+    return new UFGcpDataprocCluster(this);
+  }
+
+  /** Serialize the internal representation of the resource to the format for writing to disk. */
+  public PDGcpDataprocCluster serializeToDisk() {
+    return new PDGcpDataprocCluster(this);
+  }
+
+  /** Delete a GCP dataproc cluster referenced resource in the workspace. Currently unsupported. */
+  protected void deleteReferenced() {
+    throw new UserActionableException(
+        "Referenced resources not supported for GCP Dataproc clusters.");
+  }
+
+  /** Delete a GCP dataproc cluster controlled resource in the workspace. */
+  protected void deleteControlled() {
+    // call WSM to delete the resource
+    WorkspaceManagerServiceGcp.fromContext()
+        .deleteControlledGcpDataprocCluster(Context.requireWorkspace().getUuid(), id);
+  }
+
+  /** Retrieve cluster status from axon server */
+  public Optional<ClusterStatus> getClusterStatus() {
+    try {
+      return Optional.of(
```

Comment:
Are these values nullable? Assuming its a no. 

Applies to other get functions below

Reply:
The underlying service class can throw an exception, we intentionally catch them here and then return an empty optional. Use case is `resource list` or `resource describe`, we don't want one of the api calls to cause the whole command to fail. We still wrap the non nullable objs with optional to match return type.